### PR TITLE
Feature/administration page

### PR DESCRIPTION
**Pull request by JeanVittory**
# SUMMARY

The user and doctor profile were updated and currently, they can update their passwords.

## TO-DO

Keep migrating the endpoints from the backend

### CODE REVIEW

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx [hunk 1]
```diff
@@ -35,71 +37,83 @@ export const UserProfile = ({
...
+    setEmailStauts(email);
+    setPhoneSatus(phone);
+    setImageSelected('');
+    setEditState(editStyleOff);
+    setPrevData(dataStyleOn);
+  };
+
+  const updateData = async () => {
+    try {
+      const ACCES_TOKEN = localStorage.getItem(TOKEN);
```
Comment (ht1204) [inline src/pages/profiles/components/userProfile/UserProfile.page.jsx:94]: `ACCESS_TOKEN`

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx [hunk 2]
```diff
@@ -35,71 +37,83 @@ export const UserProfile = ({
...
+    //       headers: {
+    //         Authorization: `Bearer ${token}`,
+    //       },
+    //     });
+    //     if (response.status > 399) return errorMessage('Something went wrong');
+    //     setAppointments(response.data);
+    //   } catch (error) {
+    //     return errorMessage(error.message);
+    //   }
+    // };
```
Comment (ht1204) [inline src/pages/profiles/components/userProfile/UserProfile.page.jsx:60]: Why is this code commented? This will be removed or there is @To-Do pending?

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx [hunk 3]
```diff
@@ -164,9 +192,18 @@ export const DoctorProfile = ({
 
             <div className="qualifications__info">
               <h3>Qualifications</h3>
-              {qualifications.map((qua) => (
-                <div key={qua}>
-                  <p>{qua}</p>
+              {qualifications.map((qualification) => (
```
Comment (ht1204) [inline src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:195]: `!!qualifications.map`

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx [hunk 4]
```diff
@@ -164,9 +192,18 @@ export const DoctorProfile = ({
...
+                <div key={qualification}>
+                  <p>{qualification}</p>
+                </div>
+              ))}
+            </div>
+
+            <div className="qualifications__info">
+              <h3>Memberships</h3>
+              {memberships.map((membership) => (
```
Comment (ht1204) [inline src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:204]: `!!memberships.map`

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx [hunk 5]
```diff
@@ -36,67 +40,84 @@ export const DoctorProfile = ({
...
+    //         Authorization: `Bearer ${token}`,
+    //       },
+    //     });
+    //     if (response.status > 399) return errorMessage('Something went wrong');
+    //     setAppointments(response.data);
+    //   } catch (error) {
+    //     return errorMessage(error.message);
+    //   }
+    // };
+    // getAppointments();
```
Comment (ht1204) [inline src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:101]: Will this be removed or there is a @To-Do pending?

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx (hunk 2 above)
Comment (JeanVittory) [reply src/pages/profiles/components/userProfile/UserProfile.page.jsx:60]: there is a comment at line 45. This is the call to get the user appointments

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx [hunk 6]
```diff
@@ -35,71 +37,83 @@ export const UserProfile = ({
...
   const [emailStatus, setEmailStauts] = useState(email);
   const [phoneStatus, setPhoneSatus] = useState(phone);
+  const [isUpdating, setIsUpdating] = useState(false);
   const [appointments, setAppointments] = useState([]);
+  const [imageSelected, setImageSelected] = useState('');
   const [file, setFile] = useState(null);
-  const [newAvatar, setNewAvatar] = useState(null);
-
+  const navigate = useNavigate();
   //@Todo: assign link to backend to update actions
```
Comment (JeanVittory) [inline src/pages/profiles/components/userProfile/UserProfile.page.jsx:45]: here

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx (hunk 3 above)
Comment (JeanVittory) [reply src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:195]: i don't get the idea of use a  boolean coercion in this line. Please explain me.

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx (hunk 4 above)
Comment (JeanVittory) [reply src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:204]: here too. I'm not getting the idea of use this feature of javascript here

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx (hunk 1 above)
Comment (JeanVittory) [reply src/pages/profiles/components/userProfile/UserProfile.page.jsx:94]: Done!

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx (hunk 2 above)
Comment (ht1204) [reply src/pages/profiles/components/userProfile/UserProfile.page.jsx:61]: I consider the to-do comment would be closer to this commented code for a better understanding of your progress, no problem.

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx (hunk 3 above)
Comment (ht1204) [reply src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:192]: What I propose and this was commented with Sergio is avoiding any nullable values or errors fetching data and filling an array. In this kind of cases, this would provide a safe validation for conditional rendering, at least if you have validated the data for the array, there is no problem.

File: src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx (hunk 4 above)
Comment (ht1204) [reply src/pages/profiles/components/doctorProfile/DoctorProfile.page.jsx:201]: I explained it in comment above, it´s a short proposal enhancement, no problem.

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx (hunk 6 above)
Comment (ht1204) [reply src/pages/profiles/components/userProfile/UserProfile.page.jsx:46]: Move the comment closer to commented code, please.

File: src/pages/doctors/components/doctorDetail/DoctorDetail.doctors.jsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import './doctordetail.doctors.scss';
+
+export const DoctorDetail = () => {
+  const { id } = useParams();
+
+  const { doctorFilter, isLoading, error } = useDoctorsContext();
+  const [date, setDate] = useState(new Date());
+
+  const doctor = doctorFilter(id);
+  console.log(doctor);
```
Comment (ht1204) [inline src/pages/doctors/components/doctorDetail/DoctorDetail.doctors.jsx:17]: remove this console.log, please.

File: src/pages/doctors/components/doctorList/DoctorsList.doctors.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import { Link } from 'react-router-dom';
+import { useDoctorsContext } from '../../context/DoctorsContext';
+import { Spinner } from '@chakra-ui/react';
+import './doctorList.components.scss';
+
+export const DoctorsList = () => {
+  const { doctors, error, isLoading } = useDoctorsContext();
+  if (error) return <p>Error</p>;
+  console.log(doctors);
```
Comment (ht1204) [inline src/pages/doctors/components/doctorList/DoctorsList.doctors.jsx:9]: also here.

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx [hunk 9]
```diff
@@ -42,6 +42,7 @@ export const UserProfile = ({
   const [imageSelected, setImageSelected] = useState('');
   const [file, setFile] = useState(null);
   const navigate = useNavigate();
+
   //@Todo: assign link to backend to update actions
   useEffect(() => {
     // const token = localStorage.getItem(TOKEN);
```
Comment (ht1204) [inline src/pages/profiles/components/userProfile/UserProfile.page.jsx:48]: Will this commented code be deleted?

File: src/pages/profiles/components/userProfile/UserProfile.page.jsx [hunk 10]
```diff
@@ -213,7 +214,7 @@ export const UserProfile = ({
             </div>
           </div>
 
-          <UpdatePassword updater={updateUser} url={UPDATE_USER} />
+          {isUpdating || <UpdatePassword updater={updateUser} url={UPDATE_USER} />}
```
Comment (ht1204) [inline src/pages/profiles/components/userProfile/UserProfile.page.jsx:217]: `{isUpdating &&`

File: src/pages/profiles/components/userProfile/userProfile.page.scss [hunk 11]
```diff
@@ -150,7 +150,8 @@
   }
 }
 
-.appointments-none {
+ {
```
Comment (ht1204) [inline src/pages/profiles/components/userProfile/userProfile.page.scss:153]: why did you delete the css class?
